lib/v2/gov/nifdc/index.js

```diff
@@ -0,0 +1,85 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const params = ctx.path === '/nifdc' ? '/nifdc/bshff/ylqxbzhgl/qxggtzh' : ctx.path;
+
+    const rootUrl = 'https://www.nifdc.org.cn';
+    const currentUrl = `${rootUrl}${params}/index.html`;
+
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    let items = $('.list ul li a')
+        .slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 20)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            const link = item.attr('href');
+
+            return {
+                title: item.text(),
+                link: /^http/.test(link) ? link : `${rootUrl}/nifdc/${link.replace(/\.\.\//g, '')}`,
+                pubDate: parseDate(
+                    item
+                        .next()
+                        .text()
+                        .match(/\((.*)\)/)
+                ),
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                if (/^https:\/\/www\.nmpa\.gov\.cn\//.test(item.link)) {
+                    const browser = await require('@/utils/puppeteer')();
```

Comment:
The browser should be initiated outside Promise, which equals to opening 1 browser with n tabs.
The current code will open n browsers with 1 tab which will cost more resources and take longer to finish.